templates/_entries/search/default.twig

```diff
@@ -1,17 +1,114 @@
+{% set searchQuery = craft.app.request.getParam('q') %}
+{% set searchSection = craft.app.request.getParam('section') %}
```

Comment:
You can't query all sections as some "sections" are not actual pages, but rather collections of data for other purpose. The global features channel is one example.